pymc_marketing/mmm/base.py

```diff
@@ -511,7 +511,7 @@ def budget_allocator(
         self,
         total_budget: Optional[float] = None,
         df_estimations: Optional[pd.DataFrame] = None,
-        budget_bounds: Optional[Dict] = None,
+        budget_bounds: Optional[Dict] = {},
```

Comment:
**Small warning ;) :** it is not good practice to use mutable objects as defaults, see https://www.30secondsofcode.org/python/s/mutable-default-arguments/

You can keep it `None` and add inside the function:
```python
if budget_bounds is None:
    budget_bounds = {}
```